### PR TITLE
Update `actions/checkout` with `fetch-depth: 0`

### PR DESCRIPTION
**Pull request by Saransh-cpp**
### Describe your change:

See https://github.com/Saransh-cpp/github-stats./pull/1

cc: @cclauss

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by cclauss** [comment]
@dhruvmanila I am trying to get our [directory_writer](https://github.com/TheAlgorithms/Python/blob/master/.github/workflows/directory_writer.yml) GitHub Action to run on this pull request.  If I modify `directory_writer.yml` or add a new `.py` file shouldn't that kick off a run of that job?

**Comment by Saransh-cpp** [comment]
The workflow is not enabled for PRs from a fork -

https://github.com/TheAlgorithms/Python/blob/c9b4b8002f24a33ea49c16dff5ef9cbebbd64b1d/.github/workflows/directory_writer.yml#L3-L4

**Review by cclauss — APPROVED** [review]

File: arithmetic_analysis/junk.py
Comment (tianyizheng02) [inline arithmetic_analysis/junk.py:1]: Why is there this empty file??

**Comment by cclauss** [comment]
`arithmetic_analysis/junk.py` can be deleted.

**Comment by tianyizheng02** [comment]
See #9062

**Pull request merged**